### PR TITLE
[SSP-2719] Implement Contract Placeholder Design

### PR DESCRIPTION
**Pull request by jhbertra**
Added a placeholder view for a contract in a "starting" state.

- Refactored the contract state into an ADT that explicitly distinguishes between starting and started states.
- Refactored contract screen to remove duplicate component styling.
- Added a material circular progress indicator, as specified by the design.

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

### CODE REVIEW

File: web-common/src/Material/Progress.purs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+progressClass Progress70 = "progress-70"
+
+progressClass Progress80 = "progress-80"
+
+progressClass Progress90 = "progress-90"
+
+progressClass Progress100 = "progress-100"
+
+progressClass Indeterminate = "progress-indeterminate"
```
Comment (jhbertra) [inline web-common/src/Material/Progress.purs:40]: These changes could be cherry picked into a separate PR against master. This could be useful in other changes and projects.

File: marlowe-dashboard-client/src/Contract/View.purs [hunk 2]
```diff
@@ -1083,3 +1080,91 @@ getParty (S.IDeposit _ p _ _) = Just p
 getParty (S.IChoice (ChoiceId _ p) _) = Just p
 
 getParty _ = Nothing
+
+-------------------------------------------------------------------------------
+-- UI Layout helpers
+-------------------------------------------------------------------------------
+withBaseCss ::
```
Comment (hrajchert) [inline marlowe-dashboard-client/src/Contract/View.purs:1087]: What is the idea behind `withBaseCss`? Why not just have `div [ classNames ["grid",...]]`

**Review by hrajchert — APPROVED** [review]
LGTM

File: marlowe-dashboard-client/src/Contract/View.purs (hunk 2 above)
Comment (jhbertra) [reply marlowe-dashboard-client/src/Contract/View.purs:1087]: This idea was arrived at by refactoring and extracting patterns a few times:

Starting point:

```purescript
titleBar1 = header [ classNames [ "p-4", "border", "border-gray", "flex", "items-center", "justify-between" ] ]
  [ text "I'm title #1"
  , icon ...
  ]

...

titleBar2 = header [ classNames [ "p-4", "border", "border-gray" ] ]
  [ text "I'm title #2"
  ]

...

titleBar3 = header [ classNames [ "p-4", "border", "border-gray" ] ]
  [ text "I'm title #3"
  ]

...

body1 = section [ classNames [ "grid", "p-4", "bg-white" ] ]
  [ text "I'm body #1"
  , icon ...
  ]

...

body2 = section [ classNames [ "grid", "p-4", "bg-gray" ] ]
  [ text "I'm body #2"
  , icon ...
  ]
```

Pattern: multiple fragments of HTML using the same styles
Goal: Collect related styles into reusable helper
Refactor: extract helpers to create components with same "container" styles:

```purescript
titleBar css = header [ classNames [ "p-4", "border", "border-gray" ] <> css ]

body css = section [ classNames [ "grid", "p-4" ] <> css ]

titleBar1 = titleBar [ "flex", "items-center", "justify-between" ]
  [ text "I'm title #1"
  , icon ...
  ]

...

titleBar2 = titleBar [] [ text "I'm title #2" ]

...

titleBar3 = titleBar [] [ text "I'm title #3" ]

...

body1 = body  [ "bg-white" ]
  [ text "I'm body #1"
  , icon ...
  ]

...

body2 = body [ "bg-gray" ]
  [ text "I'm body #2"
  , icon ...
  ]
```

Pattern: Multiple helpers which differ only in the element rendered, and the base set of styles
Refactor: Extract pattern into single helper

```purescript
withBaseCss el baseCss css = el [ classNames $ baseCss <> css ]

titleBar = header `withBaseCss` [ "p-4", "border", "border-gray" ]

body = section `withBaseCss` [ "grid", "p-4" ]

titleBar1 = titleBar [ "flex", "items-center", "justify-between" ]
  [ text "I'm title #1"
  , icon ...
  ]

...

titleBar2 = titleBar [] [ text "I'm title #2" ]

...

titleBar3 = titleBar [] [ text "I'm title #3" ]

...

body1 = body  [ "bg-white" ]
  [ text "I'm body #1"
  , icon ...
  ]

...

body2 = body [ "bg-gray" ]
  [ text "I'm body #2"
  , icon ...
  ]
```

That's the idea, anyway. I think perhaps 80% of the value is in the first refactor, and the second one isn't as helpful. It might even be detrimental because of the additional layer of indirection (and I think in this example, it actually increases the overall amount of code)
